.gitlab/check.yml

```diff
@@ -0,0 +1,46 @@
+.check_skip_rust:
+  before_script:
+    - |-
+      if echo "$CI_COMMIT_TITLE" | grep -q "ci-skip-rust"; then
+        echo "Skipping rust tests due to commit title"
+        exit 0
+      fi
+
+cargo-clippy:
+  extends: .check_skip_rust
+  image: paritytech/ci-unified:bullseye-1.74.0
+  variables:
+    RUSTDOCFLAGS: '-D warnings'
+  stage: check
+  cache:
+    key: cargo-clippy
+    paths:
+    - ${CARGO_CACHE_PATH}
+  script:
+    -  SKIP_WASM_BUILD=1 cargo clippy --all-targets --locked --quiet
+
+cargo-clippy-all-features:
+  extends: .check_skip_rust
+  image: paritytech/ci-unified:bullseye-1.74.0
+  variables:
+    RUSTDOCFLAGS: '-D warnings'
+  stage: check
+  cache:
+    key: cargo-clippy-all-features
+    paths:
+      - ${CARGO_CACHE_PATH}
+  script:
+    -  SKIP_WASM_BUILD=1 cargo clippy --all-targets --all-features --locked --quiet
+
+check-rustdoc:
```

Comment:
We still need to treat warnings as errors (as it is now): `RUSTDOCFLAGS='-D warnings' cargo doc --all-features --no-deps --locked`

Reply:
Maybe there's a way to set it for all jobs in this stage?

Reply:
no stage level varible, only either global or job, choosing later